src/js/datepicker.js

```diff
@@ -84,71 +49,52 @@ DatePicker = class {
     buttonNext.textContent = "Next";
     datepickerNav.append(buttonNext);
 
-    let monthView = new CalendarMonthView(
-      this.currentMonth.monthName,
-      this.currentMonth.firstWeekdayOfMonth,
-      this.currentMonth.numberOfDays,
-      this.currentMonth.year,
-      this.currentMonth.isCurrentMonth,
-      this.currentMonth.dayNumToday
-    );
-    calendarMonthWrap.appendChild(monthView.monthView);
 
     calendarHtml.appendChild(datepickerNav);
     calendarHtml.appendChild(calendarMonthWrap);
 
     // not sure why I can't use this.currentMonthIndex directly here, comes back as undefined
     let currentMonthIndex = this.currentMonthIndex;
-    let months = this.months;
-    let yearInView = this.currentYear;
-    let monthInViewIndex = monthInViewFromAllMonths();
+    let currentYear = this.currentYear;
+    let today = this.today;
     let monthsInPast = this.numberOfMonthsInPast;
-    // let monthsInFuture = this.numberOfMonthsInFuture;
-
-    console.log(this.months);
 
-    function monthInViewFromAllMonths() {
-      let monthInViewIndex;
-      console.log(months);
+    addNewMonthViewToCalendar();
 
-      months.forEach((month, index) => {
-        if (month.year === yearInView && month.monthIndex === currentMonthIndex) {
-          monthInViewIndex = index;
-        }
-      });
+    function addNewMonthViewToCalendar() {
+      let month = new CalendarMonth(currentYear, currentMonthIndex, today);
+      let newMonthView = new CalendarMonthView(month);
+      let calendarMonth = calendarMonthWrap.querySelector('.datepicker__calendar-month');
 
-      console.log(monthInViewIndex);
-      return monthInViewIndex;
-    }
+      if (calendarMonth) {
+        calendarMonthWrap.querySelector('.datepicker__calendar-month').remove();
+      }
 
-    function addNewMonthViewToCalendar() {
-      let month = months[monthInViewIndex];
-      let newMonthView = new CalendarMonthView(
-        month.monthName,
-        month.firstWeekdayOfMonth,
-        month.numberOfDays,
-        month.year,
-        month.isCurrentMonth,
-        month.dayNumToday
-      );
-      monthView.monthView.parentNode.replaceChild(newMonthView.monthView, monthView.monthView);
-      monthView = newMonthView;
+      calendarMonthWrap.appendChild(newMonthView.createCalendarMonthHtml());
     }
 
     buttonPrev.addEventListener('click', function () {
-      if (monthInViewIndex >= currentMonthIndex - monthsInPast) {
-        monthInViewIndex--;
-        addNewMonthViewToCalendar();
+      currentMonthIndex--;
```

Comment:
so you could refactor both these in a `changeMonth(value)` function, where prev triggers `changeMonth(-1)` and next triggers `changeMonth(1)`, and then have both 0 and 11 conditions in there , would remove the duplication around `addNewMonthViewToCalendar`

Reply:
yes, that's bit better, thanks! :)